src/shogun/mathematics/linalg/ratapprox/logdet/opfunc/LogRationalApproximationCGM.cpp

```diff
@@ -42,34 +42,29 @@ void CLogRationalApproximationCGM::init()
 
 	SG_ADD((CSGObject**)&m_linear_solver, "linear_solver",
 		"Linear solver for complex systems", MS_NOT_AVAILABLE);
-
-	SG_ADD(&m_negated_shifts, "negated_shifts",
-		"Negated shifts", MS_NOT_AVAILABLE);
 }
 
 CLogRationalApproximationCGM::~CLogRationalApproximationCGM()
 {
 	SG_UNREF(m_linear_solver);
 }
 
-float64_t CLogRationalApproximationCGM::compute(SGVector<float64_t> sample)
+float64_t
+CLogRationalApproximationCGM::compute(SGVector<float64_t> sample) const
 {
 	SG_DEBUG("Entering\n");
 	REQUIRE(sample.vector, "Sample is not initialized!\n");
 	REQUIRE(m_linear_operator, "Operator is not initialized!\n");
 
 	// we need to take the negation of the shifts for this case
-	if (m_negated_shifts.vector == NULL)
-	{
-		m_negated_shifts = SGVector<complex128_t>(m_shifts.vlen);
-		Map<VectorXcd> shifts(m_shifts.vector, m_shifts.vlen);
-		Map<VectorXcd> negated_shifts(
-			m_negated_shifts.vector, m_negated_shifts.vlen);
-		negated_shifts = -shifts;
-	}
+
+	SGVector<complex128_t> negated_shifts(m_shifts.vlen);
```

Comment:
I think this is a bit inefficient.
Everytime this method is called, a new vector is allocated, content is copied and it is negated.
Compute is called many times.
I guess this is why negated shifts was a member.
But we can easily avoid this with a boolean flag in the `solve_shifted`weighted` or so?